code/modules/experisci/experiment/types/random_scanning.dm

```diff
@@ -19,3 +19,25 @@
 
 	// Fill the experiment as per usual
 	..()
+
+/datum/experiment/scanning/random/mecha_damage_scan/New(datum/techweb/techweb)
+	. = ..()
+	damage_percent = rand(15, 95)
+	//updating the description with the damage_percent var set
+	description = "Your exosuit fabricators allow for rapid production on a small scale, but the structural integrity of created parts is inferior to those made with more traditional means. Damage a few exosuits to around [damage_percent]% integrity and scan them to help us determine how the armor fails under stress."
+
+/datum/experiment/scanning/random/mecha_damage_scan/final_contributing_index_checks(datum/component/experiment_handler/experiment_handler, atom/target, typepath)
+	var/found_percent = round((target.get_integrity() / target.max_integrity) * 100)
+	return ..() && ISINRANGE(found_percent, damage_percent - 5, damage_percent + 5)
+
+/datum/experiment/scanning/random/mecha_equipped_scan/final_contributing_index_checks(datum/component/experiment_handler/experiment_handler, atom/target, typepath)
+	var/obj/vehicle/sealed/mecha/stompy = target
+	if(!istype(stompy))
+		return FALSE //Not a mech
+	if(!HAS_TRAIT(stompy,TRAIT_MECHA_CREATED_NORMALLY))
+		experiment_handler.announce_message("Scanned mech was not made by crew. Their is nothing to learn here.")
+		return FALSE //Not hand-crafted
+	if(!(stompy.equip_by_category[MECHA_L_ARM] && stompy.equip_by_category[MECHA_R_ARM]))
+		experiment_handler.announce_message("Scanned mech is missing equipment on their arms to proceed this experiment.")
```

Comment:
Please tell me if anyone wants some other logics announce why the experiment is failing.

Reply:
also: "Their is nothing to learn here", minor spelling nitpick

Reply:
Whats the spelling nitpick? your saying what is already in the code.

Reply:
That would be "There is nothing to learn here."
Which to use when out of their/they're/there can be hard shit to learn, so I getcha